spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/mqtt/MqttHandler.java

```diff
@@ -218,20 +220,36 @@ public void deliveryComplete(IMqttDeliveryToken token) {
         // todo: update or remove the audit table rows when changes have been successfully delivered to backend
     }
 
+    @Override
+    public void connectComplete(boolean reconnect, String serverURI) {
+        if (reconnect) {
+            clientConnected.onNext(true);
+            //clean session, re-subscribe
+            SpatialConnect.getInstance().getBackendService().reconnect();
+        }
+    }
+
     /**
      * An implementation of an IMqttActionListener for connecting/authenticating to the broker.
      */
     class ConnectActionListener implements IMqttActionListener {
         @Override
         public void onSuccess(IMqttToken asyncActionToken) {
             Log.d(LOG_TAG, "Connection Success!");
-            clientConnected.onNext(1);
+            DisconnectedBufferOptions disconnectedBufferOptions = new DisconnectedBufferOptions();
+            disconnectedBufferOptions.setBufferEnabled(true);
+            disconnectedBufferOptions.setBufferSize(100);
+            disconnectedBufferOptions.setPersistBuffer(false);
+            disconnectedBufferOptions.setDeleteOldestMessages(false);
```

Comment:
I think we would want to delete the old messages if the buffer fills up to avoid being a memory hog.  Consider changing this to `disconnectedBufferOptions.setDeleteOldestMessages(true);`?